lib/collection/set.js

```diff
@@ -41,31 +39,35 @@ export class SetColl extends Collection {
   }
 
   remove(item) {
-    this._removeWithoutObserver(item);
-    this._notifyRemoved([item]);
+    if (this._removeWithoutObserver(item)) {
+      this._notifyRemoved([item]);
+    }
   }
 
   _removeWithoutObserver(item) {
```

Comment:
Document the new return value
```suggestion
  /** @returns {boolean} whether the value was in the set and remove, or not. */
  _removeWithoutObserver(item) {
```